### PR TITLE
More bug fixes for Apple password store

### PR DESCRIPTION
**Pull request by colincornaby**
- Default arguments to CFStringCreateWithCStringNoCopy deallocate the source buffer on dealloc. kCFAllocatorNull prevents this.
- Creating only on set of CF ojbects in SetPassword.
- SetPassword wasn’t setting the password data on the initial store.

### CODE REVIEW

File: Sources/Plasma/FeatureLib/pfPasswordStore/pfPasswordStore_Apple.cpp [hunk 1]
```diff
@@ -85,38 +85,34 @@ bool pfApplePasswordStore::SetPassword(const ST::string& username, const ST::str
 {
     ST::string service = GetServerDisplayName();
     
-    CFStringRef accountName = CFStringCreateWithCStringNoCopy(nullptr, username.c_str(), kCFStringEncodingUTF8, nullptr);
-    CFStringRef serviceName = CFStringCreateWithCStringNoCopy(nullptr, service.c_str(), kCFStringEncodingUTF8, nullptr);
+    CFStringRef accountName  = CFStringCreateWithCStringNoCopy(nullptr, username.c_str(), kCFStringEncodingUTF8, kCFAllocatorNull);
+    CFStringRef serviceName  = CFStringCreateWithCStringNoCopy(nullptr, service.c_str(), kCFStringEncodingUTF8, kCFAllocatorNull);
+    CFDataRef   passwordData = CFDataCreate(nullptr, (const UInt8*)password.c_str(), password.size());
```
Comment (Hoikas) [inline Sources/Plasma/FeatureLib/pfPasswordStore/pfPasswordStore_Apple.cpp:90]: Does this need to be released?

Comment (colincornaby) [reply Sources/Plasma/FeatureLib/pfPasswordStore/pfPasswordStore_Apple.cpp:90]: It does! Thanks.

**Comment by colincornaby** [comment]
The run has failed, it looks like there was a vcpkg timeout. I don't think I have access to restart the workflow.

**Comment by Deledrius** [comment]
> The run has failed, it looks like there was a vcpkg timeout. I don't think I have access to restart the workflow.

Looks like a successful run now. :)

**Review by Hoikas — APPROVED** [review]

**Pull request merged**